Fix depwarns on Julia 0.6

ararslan opened the pull request.

ararslan commented:
Weird, unrelated Travis failure on Julia 0.4. Will close and reopen to see if it goes away.

mlubin commented:
Travis failure very likely related: https://github.com/JuliaDiff/DualNumbers.jl/pull/50/commits/264d32840e3624433a08433458aa789060014e10

No objection here to dropping support for 0.4 though.

ararslan commented:
It was failing on all of Linux without that change. I commented out the overridden test script to see if it would get it working, and it does, for everything _but_ 0.4 on Linux...

ararslan commented:
As it turns out, it's a bug with Travis. I've pushed a fix to the Travis YAML that works around it. Things should be fine on 0.4 now.

ararslan commented:
I've reorganized the commits to group related changes. The Travis fixes are now a single separate commit. I've also added the PkgEval badges to the README and updated the Travis badge to point specifically to the master branch. (Currently it shows the status from the most recent build, which may not be relevant to master at all.)

jrevels commented on .travis.yml:
```diff
@@ -9,8 +9,11 @@ julia:
+  depth: 99999
+#script:
+#  - if [[ -a .git/shallow ]]; then git fetch --unshallow; fi
+#  - julia -e 'Pkg.clone(pwd()); Pkg.build("DualNumbers"); Pkg.test("DualNumbers"; coverage=true)'
```
Is it reasonable to leave the actual testing portion of the script here uncommented? Will the Travis workaround still work in that case? It seems Travis is running a default Julia test script, but might as well be explicit (it took me a while to understand what was going on here).

ararslan replied:
As I understand it, the preferred option is to have this portion commented; it's in [the Example repo](https://github.com/JuliaLang/Example.jl/blob/master/.travis.yml#L11) and it's [generated by PkgDev](https://github.com/JuliaLang/PkgDev.jl/blob/4fcf2f5f45364f5bbfd9b3dd644f17f405aa259d/src/generate.jl#L233). The advantage is that if ever the default changes, it's best not to override it, because a change to the default might be reflective of a change in Travis itself that wouldn't be caught with this script.

The workaround for the Travis depth bug should work in either case.

ararslan replied:
That said, it's totally up to you; I'll do whatever you think is best.

jrevels replied:
This seems fine then - I didn't realize this is the new default for the `.travis.yml` generated by `PkgDev`.

ararslan commented:
Thanks, Miles and Jarrett!

ararslan commented:
Any chance someone could do a patch release to fix the depwarns?
